### PR TITLE
Add release script pt 164978502

### PR DESCRIPTION
**Pull request by valerifilipov**
Please specify own repos with `NODE_REPO` and `PROTOCOL_REPO`
https://github.com/aeternity/infrastructure/blob/f8d897445aca51607cacfdd30abab6d00ce7d084/scripts/release.sh#L16-L17
Please comment
Example usage and output:
```bash
scripts/release.sh 2.5.4 start Test release --node-commitish=test --protocol-commitish=protocol-test
Creating pre-release aeternity-node-v2.5.4 Test release in valerifilipov/repo2 from commitish protocol-test
Pre-release created
Creating pre-release v2.5.4 Test release in valerifilipov/repo1 from commitish test
Pre-release created
```
```bash
scripts/release.sh 2.5.4 finish
Publishing release aeternity-node-v2.5.4 for repo: valerifilipov/repo2
aeternity-node-v2.5.4 released sucessfully in valerifilipov/repo2
Publishing release v2.5.4 for repo: valerifilipov/repo1
v2.5.4 released sucessfully in valerifilipov/repo1
```

### CODE REVIEW

File: scripts/relase.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+# should be run as:
+# ./release.sh x.x.x start|end description
```
Comment (lucafavatella) [inline scripts/relase.sh]: release.sh or relase.sh?

Comment (valerifilipov) [reply scripts/relase.sh]: :D sorry, fixed it

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+# should be run as:
+# ./release.sh x.x.x start|end description
+set -eo pipefail
+
+USAGE="Usage:
+${0} x.x.x start|end description, E.g. to start a release run:
+${0} 2.4.0 start Maintainance release
```
Comment (lucafavatella) [inline scripts/release.sh]:
```suggestion
${0} 2.4.0 start Maintenance release
```

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+# ./release.sh x.x.x start|end description
+set -eo pipefail
+
+USAGE="Usage:
+${0} x.x.x start|end description, E.g. to start a release run:
+${0} 2.4.0 start Maintenance release
+then to finish the release run:
+${0} 2.4.0 end"
+
+protocol_repo=valerifilipov/repo2
```
Comment (dincho) [inline scripts/release.sh]: let's make this environment variables with defaults:

```
PROTOTOL_REPO=${PROTOTOL_REPO:-aeternity/protocol}
```

File: scripts/release.sh [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+# should be run as:
+# ./release.sh x.x.x start|end description
+set -eo pipefail
+
+USAGE="Usage:
+${0} x.x.x start|end description, E.g. to start a release run:
```
Comment (dincho) [inline scripts/release.sh]: @lucafavatella is start/finish more appropriate wording ?

File: scripts/release.sh [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  "body": "$1",
+  "draft": false,
+  "prerelease": $prerelease
+}
+EOF
+}
+
+curl_headers=(
+    '-H' "Accept: application/vnd.github.v3+json"
+    '-H' "Authorization: token $VAULT_GITHUB_TOKEN"
```
Comment (dincho) [inline scripts/release.sh]: This shouldn't coupled to Vault. Use generic variable and make sure it's exported in the import-secrets.
Also use required variables where appropriate, like this case it should be `${VAULT_GITHUB_TOKEN:?}`

File: scripts/release.sh [hunk 6]
```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
```
Comment (dincho) [inline scripts/release.sh]: this script should take target comitish as parameters as well (commit candidates)

File: scripts/release.sh (hunk 3 above)
Comment (valerifilipov) [reply scripts/release.sh]: Addressed in f8d8974. Will change defaults in last commit

File: scripts/release.sh (hunk 4 above)
Comment (valerifilipov) [reply scripts/release.sh]: changed in 865ce23

File: scripts/release.sh (hunk 5 above)
Comment (valerifilipov) [reply scripts/release.sh]: Addressed in 886247f

File: scripts/release.sh (hunk 6 above)
Comment (valerifilipov) [reply scripts/release.sh]: Added in 0424292

File: scripts/release.sh [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+                        ;;
+                    *)
+                        description+=(${arg})
+                        ;;
+                esac
+            done
+            if [[ -z "$description" ]]; then
+                usage_exit
+            fi
+            protocol_commitish=${protocol_commitish:-master}
```
Comment (dincho) [inline scripts/release.sh]: Could you please move this on the top so the defaults are clearly visible

File: scripts/release.sh [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+#check for release version
+if [[ -n "$1" && "$1" =~ ^([0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)*)$ ]]; then
+    version=${1}
+    shift
+else
+    usage_exit
+fi
+
+protocol_release=aeternity-node-v$version
```
Comment (dincho) [inline scripts/release.sh]: please move all (non local) variables/constants at the top

File: scripts/release.sh [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+${0} x.x.x start|finish description [--protocol-commitish=master] [--node-commitish=master], e.g. to start a release run:
+${0} 2.4.0 start Maintenance release
+then to finish the release run:
+${0} 2.4.0 finish"
+
+usage_exit() {
+    echo -e "$USAGE" >&2; exit 1
+}
+
+protocol_repo=${PROTOCOL_REPO:-valerifilipov/repo2}
```
Comment (dincho) [inline scripts/release.sh]: Leftovers?

Should be aeternity/protocol and aeternity/aeternity

File: scripts/release.sh [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+                esac
+            done
+            if [[ -z "$description" ]]; then
+                usage_exit
+            fi
+            description=${description[@]}
+            echo "Creating pre-release ${protocol_release:?} $description in $protocol_repo from commitish $protocol_commitish"
+            create_release $protocol_repo ${protocol_release:?} $protocol_commitish "$description"
+            echo "Creating pre-release ${node_release:?} $description in $node_repo from commitish $node_commitish"
+            create_release $node_repo ${node_release:?} $node_commitish "$description"
```
Comment (lucafavatella) [inline scripts/release.sh]: Can it also add text in release e.g.:
https://github.com/aeternity/aeternity/releases/tag/v2.5.0
> Please see the [release notes](https://github.com/aeternity/aeternity/blob/v2.5.0/docs/release-notes/RELEASE-NOTES-2.5.0.md).

File: scripts/release.sh [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+                        description+=(${arg})
+                        ;;
+                esac
+            done
+            if [[ -z "$description" ]]; then
+                usage_exit
+            fi
+            description=${description[@]}
+            echo "Creating pre-release ${protocol_release:?} $description in $protocol_repo from commitish $protocol_commitish"
+            create_release $protocol_repo ${protocol_release:?} $protocol_commitish "$description"
```
Comment (lucafavatella) [inline scripts/release.sh]: Can it also add text in release e.g.:
https://github.com/aeternity/protocol/releases/tag/aeternity-node-v2.5.0
> aeternity-node-v2.5.0

Comment (valerifilipov) [reply scripts/release.sh]: it adds the text specified after the version, e.g.: `release.sh 2.5.4 start Minerva maintenance release`

Comment (lucafavatella) [reply scripts/release.sh]: I mean not the title but the body.

**Pull request merged**